compose/foundation/foundation/src/desktopTest/kotlin/androidx/compose/foundation/ScrollbarTest.kt

```diff
@@ -650,11 +695,12 @@ class ScrollbarTest {
                 }
             }
 
+            @Suppress("DEPRECATION")
```

Comment:
> I will probably also add tests (or a way to test) whether implementations of the old ScrollbarAdapter interface still work correctly.

It seems we still test the old functions here? How does `thumb size on scrollbar smaller than viewport` pass?

Reply:
It's actually using the new implementation behind the scenes.

`ScrollbarAdapter(state)` returns `NewAdapterAsOld(NewScrollbarAdapter(scrollState))` and then when you pass it to `Scrollbar`, it just unwraps the new adapter:
```
private fun ScrollbarAdapter.asNewAdapter(trackSize: Int) =
    if (this is NewAdapterAsOld)
        this.newAdapter  // Just unwrap
    else
        OldAdapterAsNew(this, trackSize)
```

So existing code will automatically use the new implementation, and will enjoy the fix (unless it actually implements the old `ScrollbarAdapter` itself).

Reply:
It also means that I need to come up with a good way to test whether the old adapter really works when it's wrapped into `OldAdapterAsNew`.

Reply:
Nice trick :).

Let's change code to the new API though, it is always better to test what users will use directly.

Reply:
Now testing both new and old code.